warning: class 'HANumeric' is implicitly friends with itself

I am getting this warning when importing the library:
```
In file included from /Users/REDACTED/arduino/libraries/home-assistant-integration/src/device-types/HASensorNumber.h:5,
                 from /Users/REDACTED/Project/Project.ino:68:
/Users/REDACTED/arduino/libraries/home-assistant-integration/src/device-types/../utils/HANumeric.h:232:18: warning: class 'HANumeric' is implicitly friends with itself
  232 |     friend class HANumeric;
....
Multiple times
```

--- a/src/utils/HANumeric.h
+++ b/src/utils/HANumeric.h
@@ -228,8 +228,6 @@ private:
     uint8_t _precision;
 
     explicit HANumeric(const int64_t value);
-
-    friend class HANumeric;
 };
 
 #endif

